scripts/issue/1133/upkeep_manager_integration.py

```diff
@@ -0,0 +1,120 @@
+upkeep_manager_snap = GreatApeSafe(r.badger_wallets.upkeep_manager)
+
+# contract
+upkeep_manager = techops.contract(r.badger_wallets.upkeep_manager)
+
+
+def initial_setup():
+    """
+    carries the following actions: cancels upkeeps from both registries,
+    sends techops link funds to manager, init the base upkeep in the manager,
+    updates watchlist, updates registry used in the gas station to latest and
+    register the gas station so it sends 1 eth to the manager
+    """
+    techops.init_badger()
+    techops.init_chainlink()
+
+    # snaps
+    techops.take_snapshot([techops.chainlink.link])
+    upkeep_manager_snap.take_snapshot([techops.chainlink.link])
+
+    initial_rounds_top_up = upkeep_manager.roundsTopUp()
+
+    # 1. Send links funds to allow to init base upkeep job
+    link_bal = techops.chainlink.link.balanceOf(techops)
+    techops.chainlink.link.transfer(upkeep_manager, link_bal)
+
+    # NOTE: reduce the min rounds so we can initialize it, since techops only has 90 LINK
+    upkeep_manager.setRoundsTopUp(2)
+
+    # NOTE: while testing with foundry, setting up on this figure have not seen any revert
+    # on top-up actions or withdrawal of funds from the keepers.
+    # ref of gas snap: https://github.com/Badger-Finance/badger-avatars/blob/main/.gas-snapshot#L166-L176
+    upkeep_manager.initializeBaseUpkeep(1_000_000)
+
+    upkeep_manager_snap.print_snapshot()
+
+    # 2. Cancel upkeep in old and new registry to migrate ownership to the upkeep manager
+    for name, upkeep_id in members.items():
+        if name == "GasStationExact":
+            techops.chainlink.keeper_registry_v1_1.cancelUpkeep(upkeep_id)
+        else:
+            techops.chainlink.keeper_registry.cancelUpkeep(upkeep_id)
+
+    # 3. Update gas station
+    techops.badger.set_gas_station_watchlist()
+
+    # 4. Set `s_keeperRegistryAddress` to latest cl registry
+    techops.badger.station.setKeeperRegistryAddress(techops.chainlink.keeper_registry)
+
+    # 5. Add gas st in this tx with the left-over link funds to auto-fund the manager in next block with eth
+    upkeep_manager.addMember(techops.badger.station, "GasStationExact", 400_000, 0)
+    upkeep_manager.setRoundsTopUp(initial_rounds_top_up)
+
+    upkeep_manager_snap.print_snapshot()
+    techops.post_safe_tx(call_trace=True)
+
+
+def register_members_in_manager():
+    techops.init_chainlink()
+
+    for name, upkeep_id in members.items():
+        if name == "GasStationExact":
+            # NOTE: moving forward makes more sense to have links funds in upkeep manager
+            techops.chainlink.keeper_registry_v1_1.withdrawFunds(
+                upkeep_id, upkeep_manager
+            )
+        else:
+            (
+                member_address,
+                gas_limit,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+            ) = techops.chainlink.keeper_registry.getUpkeep(upkeep_id)
```

Comment:
can do something like:
`member_address, gas_limit = techops.chainlink.keeper_registry.getUpkeep(upkeep_id)[0:2]`
so that the `_` don't take up 6 lines

Reply:
thats is pretty neat, added in 230c4cff338b5c1002816a4722a45a0f54ef1807